plugins/workflow/workflow_pool.go

```diff
@@ -77,20 +87,20 @@ func (pool *workflowPool) Start(ctx context.Context, temporal temporal.Temporal)
 }
 
 // Destroy stops all temporal workers and application worker.
-func (pool *workflowPool) Destroy(ctx context.Context) {
+func (pool *workflowPool) Destroy(ctx context.Context) error {
 	for i := 0; i < len(pool.tWorkers); i++ {
 		pool.tWorkers[i].Stop()
 	}
 
-	// todo: pass via event callback
-	//if err := pool.worker.Stop(ctx); err != nil {
-	//	pool.events.register()
-	//}
+	if err := pool.worker.Stop(ctx); err != nil {
+		return err
```

Comment:
wrap error with spiral errors package (I mention this one, but please, apply on all occurrences)